src/coreComponents/physicsSolvers/fluidFlow/CompositionalMultiphaseBase.cpp

```diff
@@ -1566,96 +1569,96 @@ void CompositionalMultiphaseBase::resetViews( MeshLevel & mesh )
   }
 
   {
-    using keys = MultiFluidBase::viewKeyStruct;
     using namespace constitutive::multifluid;
+    using namespace extrinsicMeshData::multifluid;
 
     m_phaseVisc.clear();
-    m_phaseVisc = elemManager.constructMaterialArrayViewAccessor< real64, 3, LAYOUT_PHASE >( keys::phaseViscosityString(),
+    m_phaseVisc = elemManager.constructMaterialArrayViewAccessor< real64, 3, LAYOUT_PHASE >( phaseViscosity::key(),
```

Comment:
We would also benefit from a version of `constructMaterialArrayViewAccessor` that takes the trait struct as parameter (and forwards it to one of existing versions). Can be named `constructMaterialExtrinsicAccessor` or something (although honestly these names are getting slightly out of hand).

Reply:
I added a `constructMaterialExtrinsicAccessor` in `ElementRegionManager`. If we propagate `extrinsicData` everywhere, some of the functions in there will become unnecessary, and at that point we can hopefully remove a few and clean up `ElementRegionManager`.

(Also, I removed at these calls from `CompositionalMultiphaseBase.cpp` in #1696)